### ISSUE TITLE
Error after converting TemplateStringsArray to string

### ISSUE THREAD

**Issue by DiriectorDoc** [microsoft/TypeScript#48756]
Say I have a function that I would like to have this behaviour:

```js
tagTrim("   Hello  ") // → "Hello"
tagTrim`   Hello  `   // → "Hello"
```
The function looks like this:
```ts
function tagTrim(str: string | TemplateStringsArray) {
    if(Array.isArray(str)){
        str = str[0]
    }
    // Property 'trim' does not exist on type 'string | TemplateStringsArray'.
    return str.trim()
}
```
If I simply ignore the error and compile it to JS, it works every time. But it seems TS has a problem seeing that `if(Array.isArray(str)){ str = str[0] }` will always convert a TemplateStringsArray to a string.

My first assumption would be that `Array.isArray(str) /*true*/` will assume `str` is an Array<any>, but if that's the case, then `str = str[0]` should throw an error too, but it doesn't.

So if it knows that `str = str[0]` is allowed, it should know that everything after it, `str` is now a string.

**Comment by RyanCavanaugh** [comment]
To expedite the triage process, we need everyone to follow the issue template and instructions.

When you clicked "Create New Issue", the issue form was pre-populated with a template and some instructions. We need you to read those instructions *completely*, follow them, and then fill in *all* the fields in that template.

We are not able to assist with issues that don't follow the template instructions as they represent a significantly larger amount of work compared to issues which are correctly specified. Thank you for understanding.
